tests/functional/test-no-offer.sh

```diff
@@ -0,0 +1,129 @@
+#!/bin/bash
+
+project_id=75681640118a4890b3da0d106eae8af7 # test1 uuid
+tmpfile=$(mktemp ./tmpXXXXXX)
+start=$(date +%Y-%m-%d)
+end=$(date -d "+5 days" +%Y-%m-%d)
+nodefile=$(mktemp /tmp/nodes/XXXX)
+errfile=$(mktemp ./errXXXXXX)
+node_uuid=$(echo $nodefile | sed 's/\/tmp\/nodes\///')
+
+trap "rm -f $nodefile $tmpfile $errfile" EXIT
+
+# Create dummy node
+
+cat <<EOF > /tmp/nodes/$node_uuid
+{
+    "project_owner_id": "$project_id",
+    "server_config": {
+        "new attribute XYZ": "This is just a sample list of free-form attributes used for describing a server.",
+        "cpu_type": "Intel Xeon",
+        "cores": 16,
+        "ram_gb": 512,
+        "storage_type": "samsung SSD",
+        "storage_size_gb": 204
+    }
+}
+EOF
+
+#######################
+## LEASE CREATE TEST ##
+#######################
+
+# Tests that an owner can create a lease without a preexisting offer
+
+echo "LEASE CREATE TEST"
+
+openstack --os-cloud test1 esi lease create \
+  $node_uuid \
+  test1-subproject \
+  --start-time $start \
+  --end-time $end \
+  --resource-type dummy_node \
+  -f shell > $tmpfile \
+  || { ec=$?; echo "ERROR: failed to create lease" >&2; exit $ec; }
+
+cat $tmpfile
+. $tmpfile
+
+echo "LEASE CREATE TEST SUCCEEDED"
+
+#######################
+## LEASE DELETE TEST ##
+#######################
+
+# Tests that an owner can delete a lease on a node they own
+
+echo "LEASE DELETE TEST"
+
+openstack  --os-cloud test1 esi lease delete $uuid \
+  || { ec=$?; echo "ERROR: failed to delete lease" >&2; exit $ec; }
+
+echo "LEASE DELETE TEST SUCCEEDED"
+
+####################################
+## OFFER CREATE INVALID USER TEST ##
```

Comment:
This test requires an offer to be created, do you think it could be placed in test-no-lessee-id.sh?

Reply:
This doesn't require an offer to be created, it creates an offer. If I put it in the other script it'd have a time conflict with the offer in that script and might not give the error I want.

Reply:
I see!